uaclient/contract.py

```diff
@@ -492,3 +531,48 @@ def get_contract_information(cfg: UAConfig, token: str) -> Dict[str, Any]:
     """Query contract information for a specific token"""
     client = UAContractClient(cfg)
     return client.request_contract_information(token)
+
+
+def is_contract_changed(cfg: UAConfig) -> bool:
+    orig_token = cfg.machine_token
+    orig_entitlements = cfg.entitlements
+    machine_token = orig_token.get("machineToken", "")
+    contract_id = (
+        orig_token.get("machineTokenInfo", {})
+        .get("contractInfo", {})
+        .get("id", "")
+    )
+    contract_client = UAContractClient(cfg)
+    resp = contract_client.get_updated_contract_info(
+        machine_token, contract_id
+    )
+    try:
+        resp_expiry = (
+            resp.get("machineTokenInfo", {})
+            .get("contractInfo", {})
+            .get("effectiveTo", "0001-01-01T00:00:00+00:00")
```

Comment:
I think this default would make it so we never fall back to `cfg.contract_expiry_datetime`. Maybe we should default to `None` and check for None before trying to parse it?

Reply:
I included this in my suggestion below

Reply:
Took me a minute here to understand, but I got it. Yes, makes sense.